api/src/modules/projects/projects.controller.ts

```diff
@@ -68,12 +69,19 @@ export class ProjectsController {
   @Get(':projectId/features')
   async findAllGeoFeaturesForProject(
     @ProcessFetchSpecification() fetchSpecification: FetchSpecification,
-    @Param() params: { projectId: string },
+    @Param() params: { projectId: string; featureClassAndAliasFilter?: string },
+    @Query('q') featureClassAndAliasFilter: string,
   ): Promise<GeoFeatureResult> {
+    /**
+     * @debt Add proper typing in nestjs-base-service to allow adding query parameters in InfoDTO
+     */
+
+    params.featureClassAndAliasFilter = featureClassAndAliasFilter;
     const results = await this.geoFeaturesService.findAllPaginated(
       fetchSpecification,
       { params },
     );
+
```

Comment:
for this, see comments by myself and Kamil in this PR's general discussion if you want to refactor this a bit.

Reply:
So I decided to let that query param as is for simplicity and just add it to params object applying immutability according to @kgajowy 's suggestion